app/src/main/java/com/futurice/freesound/feature/audio/ExoPlayerAudioPlayer.java

```diff
@@ -86,26 +88,34 @@
     @Override
     public void init() {
         playerStateDisposable
-                .set(toggleUrlStream.concatMap(url -> observableExoPlayer
+                .set(toggleSourceStream.concatMap(playbackSource -> observableExoPlayer
                         .getExoPlayerStateOnceAndStream()
                         .take(1)
-                        .map(exoPlayerState -> toToggleAction(url, exoPlayerState))
-                        .doOnNext(action -> handleToggleAction(action, url)))
-                                    .subscribe(Functions.nothing1(),
-                                               e -> Timber.e(e, "Fatal error toggling playback")));
+                        .map(exoPlayerState -> toToggleAction(playbackSource, exoPlayerState))
+                        .doOnNext(action -> handleToggleAction(action, playbackSource)))
```

Comment:
Not super happy about doOnNext here. Cannot you somehow put it in the subscription?